src/main/scala/com/databricks/labs/overwatch/pipeline/BronzeTransforms.scala

```diff
@@ -552,45 +552,36 @@ trait BronzeTransforms extends SparkSessionWrapper {
+            .withColumn("spark_conf", SchemaTools.structToMap(scrubbedDF, "spark_conf"))
+            .withColumn("spark_env_vars", SchemaTools.structToMap(scrubbedDF, "spark_env_vars"))
+            .withColumn(s"aws_attributes", SchemaTools.structToMap(scrubbedDF, s"aws_attributes"))
+            .withColumn(s"azure_attributes", SchemaTools.structToMap(scrubbedDF, s"azure_attributes"))
+            .withColumn(s"gcp_attributes", SchemaTools.structToMap(scrubbedDF, s"gcp_attributes"))
             .withColumn("organization_id", lit(config.organizationId))
+            .drop("spec")
             .verifyMinimumSchema(clusterSnapMinimumSchema)
 
-          val explodedDF = finalDF
-            .withColumnRenamed("custom_tags", "custom_tags_old")
-            .selectExpr("*", "spec.custom_tags")
-
-          val normalizedDf = explodedDF.withColumn("custom_tags", SchemaTools.structToMap(explodedDF, "custom_tags"))
-
-          // Replace the custom_tags field inside the spec struct with custom_tags outside of spec column
-          val updatedDf = normalizedDf.schema.fields.find(_.name == "spec") match {
-            case Some(field) =>
-              field.dataType match {
-                case structType: StructType =>
-                  // Create a new struct expression, replacing the specified field with the new column
-                  val newFields = structType.fields.map { f =>
-                    if (f.name.equalsIgnoreCase("custom_tags")) {
-                      col("custom_tags").as("custom_tags") // Replace with new column if names match
-                    } else {
-                      col(s"spec.${f.name}") // Keep existing fields as is
-                    }
-                  }
-                  // Update the DataFrame with the new struct replacing the old one
-                  normalizedDf.withColumn("spec", struct(newFields: _*))
-                case _ => normalizedDf // No action if the specified structColName is not a struct type
-              }
-            case None => normalizedDf // No action if the specified structColName does not exist
-          }
-
-          updatedDf.drop("custom_tags")
-            .withColumnRenamed("custom_tags_old", "custom_tags")
+//          val finalDF = df.schema.fields.find(_.name == "spec") match {
+//            case Some(field) =>
+//              field.dataType match {
+//                case structType: StructType =>
+//                  val newFields = structType.fields.map { f =>
+//                    f.dataType match {
+//                      case _: StructType => SchemaTools.structToMap(df, s"spec.${f.name}").as(f.name)
+//                      case _ => col(s"spec.${f.name}").as(f.name)
+//                    }
+//                  }
+//                  df.withColumn("spec", struct(newFields: _*))
+//                case _ => df
+//              }
+//            case None => df
+//          }
+
+          df
+            .verifyMinimumSchema(clusterSnapMinimumSchema)
```

Comment:
why we have added `verifyMinimumSchema(clusterSnapMinimumSchema)` twice ?

Reply:
@aman-db  Done